UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs

```diff
@@ -298,26 +320,40 @@ private static void ApplyNewSpawnRoleToBody( GameObject body, Occupation request
 			PlayerScript.characterSettings = character;
 		}
 
-		//Character attributes
-		var playerSprites = body.GetComponent<PlayerSprites>();
-		if (playerSprites)
+		try
 		{
-			// This causes body parts to be made for the species, will cause death if body parts are needed and
-			// CharacterSettings is null
-			var toUseCharacterSettings = requestedOccupation.UseCharacterSettings ? character : null;
-			if (requestedOccupation.CustomSpeciesOverwrite != null)
+			//Character attributes
+			var playerSprites = body.GetComponent<PlayerSprites>();
+			if (playerSprites)
 			{
-				character.Species = requestedOccupation.CustomSpeciesOverwrite.name;
-				playerSprites.RaceOverride = requestedOccupation.CustomSpeciesOverwrite.name;
-			}
+				// This causes body parts to be made for the species, will cause death if body parts are needed and
+				// CharacterSettings is null
+				var toUseCharacterSettings = requestedOccupation.UseCharacterSettings ? character : null;
+				if (requestedOccupation.CustomSpeciesOverwrite != null)
+				{
+					character.Species = requestedOccupation.CustomSpeciesOverwrite.name;
+					playerSprites.RaceOverride = requestedOccupation.CustomSpeciesOverwrite.name;
+				}
 
 
-			playerSprites.OnCharacterSettingsChange(toUseCharacterSettings);
-		}
+				playerSprites.OnCharacterSettingsChange(toUseCharacterSettings);
+			}
 
+		}
+		catch (Exception e)
+		{
+			Logger.LogError(e.ToString());
+		}
 
-		//determine where to spawn them
-		physics.AppearAtWorldPositionServer(GetSpawnPointForOccupation(requestedOccupation));
+		try
+		{
+			//determine where to spawn them
+			physics.AppearAtWorldPositionServer(GetSpawnPointForOccupation(requestedOccupation));
```

Comment:
It would be better if there was also some fallback behavior so players dont get stuck half-way through

Reply:
yeah,  What happens if that Errors though?

Reply:
if that happens, you make it delete the game

Reply:
you log the error first, then try the fallback logic.
the game shouldn't error if you're trying to spawn someone at arrivals since that's technically the default always; but if it does fail then rip i guess, means there's a bigger issue.